crates/matrix-sdk-ui/src/timeline/inner/state.rs

```diff
@@ -794,7 +811,7 @@ pub(in crate::timeline) struct TimelineInnerMetadata {
     /// timeline items (to allow efficient pushing and popping).
     ///
     /// Private because it's not needed by `TimelineEventHandler`.
-    back_pagination_tokens: Vec<(OwnedEventId, String)>,
+    back_pagination_tokens: VecDeque<(OwnedEventId, String)>,
```

Comment:
I kept the previous order, i.e. oldest events last, but since we are using a VecDeque now, for consistency, we probably should use the same order as the others, with newest events last.